nixos/modules/services/continuous-integration/jenkins/default.nix

```
@@ -215,10 +230,23 @@ in {
       '';
 
       postStart = ''
-        until [[ $(${pkgs.curl.bin}/bin/curl -L -s --head -w '\n%{http_code}' http://${cfg.listenAddress}:${toString cfg.port}${cfg.prefix} | tail -n1) =~ ^(200|403)$ ]]; do
+        until [[ $(${pkgs.curl.bin}/bin/curl -L -s --head -w '\n%{http_code}' ${jenkinsUrl} | tail -n1) =~ ^(200|403)$ ]]; do
           sleep 1
         done
-      '';
+      '' + (lib.optionalString (cfg.withCli) ''
+
+        ${pkgs.curl.bin}/bin/curl -L -s ${jenkinsUrl}/jnlpJars/jenkins-cli.jar -o ${cfg.home}/jenkins-cli.jar
+
+        cat << __EOF__ > ${cfg.home}/jenkins-cli.sh
+        #!${pkgs.bash}/bin/bash
+        ${pkgs.jdk11}/bin/java -jar ${cfg.home}/jenkins-cli.jar -s ${jenkinsUrl} \$@
```

```suggestion
        #!/bin/sh
        exec ${pkgs.jdk11}/bin/java -jar ${cfg.home}/jenkins-cli.jar -s ${jenkinsUrl} "\$@"
```

I dont think you need `bash` for this. Also the `$@` should better be quoted. I haven't tested it.